app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -74,11 +85,60 @@ companion object {
         setSearchActivityTextWatcher()
         clickOnButtonDoneSystemKeyboard()
         onUpdateButtonClickListener()
-        trackAdapter = TrackAdapter(iTunesTrack)
-        recycleViewTrack.adapter = trackAdapter
+        initizlizedComponents()
+        setOnEditTextFocusLisneter()
+        readOnHistoryTrackList()
+        clickOnClearSearchHistoryButton()
     }
 
-    fun initializedViewElementSearchActivity() {
+    private fun initizlizedComponents() {
+        historySharedPreferences = getSharedPreferences(SHARED_PREFERENCES_HISTORY_SEARCH_FILE_NAME, MODE_PRIVATE)//инициализируем экземпляр SP
+        classHistorySearch = TrackSearchHistory(historySharedPreferences)
+        trackAdapter = TrackAdapter(iTunesTrack) {//адаптер для текущего поискового запроса
+            addNewTrackInTrackHistory(it)
+        }
+        recycleViewTrack.adapter = trackAdapter//устанавливаем для RecyclerView текущего результата поиска адаптер
+        trackHistoryAdapter = TrackAdapter(iTunesTrackSearchHistory) {
+            showToast(it)
+        }//адптер для истории поиска
+        recyclerViewHistoryTrack.adapter=trackHistoryAdapter//устанавливаем для RecyclerView истории поиска адаптер
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun readOnHistoryTrackList() {//читаем из sheared preferences файла историю поиска
+        iTunesTrackSearchHistory.addAll(classHistorySearch.getTrackArrayFromShared())
+        trackHistoryAdapter.notifyDataSetChanged()
+    }
+
+    fun showToast(track: Track) {//просто тест на кликл истории поиска
+        Toast.makeText(this@SearchActivity, "Нажата в истории - ${track.artistName}, ${track.trackName}",Toast.LENGTH_SHORT).show()
+    }
+
+
+    private fun addNewTrackInTrackHistory(newTrack: Track) {
+        val temporaryTrackArray = ArrayList<Track>()
+        temporaryTrackArray.addAll(classHistorySearch.getTrackArrayFromShared())//получаем десериализованный список треков из SP во временный лист
+        if (temporaryTrackArray.isEmpty()) {//если он пуст, то просто добавляем новый трек на нулевой индекс
+            iTunesTrackSearchHistory.add(0,newTrack)
+        } else if (temporaryTrackArray.isNotEmpty()) {
+            if (temporaryTrackArray.size == 11) {
+                temporaryTrackArray.removeAt(10)
+            }
+            val iterator: MutableIterator<Track> = temporaryTrackArray.iterator()
+            while (iterator.hasNext()) {
+                val currentTrack = iterator.next()
+                if (currentTrack.trackId == newTrack.trackId) {
+                    iterator.remove()
+                }
+            }
```

Comment:
Можно заменить одной строкой  
```suggestion
            temporaryTrackArray.removeIf { it.trackId == newTrack.trackId }
```